bokehjs/src/lib/core/util/spatial.ts

```diff
@@ -51,18 +43,55 @@ class _FlatBush extends FlatBush {
           continue
         }
 
+        const index = this._indices[pos >> 2] | 0
+
         if (nodeIndex < this.numItems * 4) {
-          results.set(index) // leaf item
+          nodeFunction(index, {x0: nodeMinX, y0: nodeMinY, x1: nodeMaxX, y1: nodeMaxY})
         } else {
           queue.push(index) // node; add it to the search queue
         }
+
       }
 
       nodeIndex = queue.pop()
     }
+  }
+
+  search_indices(minX: number, minY: number, maxX: number, maxY: number): Indices {
+    const result = new Indices(this.numItems)
+
+    function fn(result: Indices) {
+      return (index: number, _node: Rect) => { result.set(index) }
+    }
+    this.search_apply(minX, minY, maxX, maxY, fn(result))
+
+    return result
```

Comment:
Maybe I'm missing something, but all of this complexity is unnecessary:
```suggestion
    const result = new Indices(this.numItems)
    this.search_apply(minX, minY, maxX, maxY, (index) => result.set(index))
    return result
```